### DISCUSSION TITLE
Get Window element of other running WPF app

### DISCUSSION THREAD

**Discussion by samberk1** [snoopwpf/snoopwpf#368] [Q&A] [answered]
Hi

I'm trying to implement something similar to the Snoop app, to be able to read properties from another WPF app but `HwndSource.FromHwnd` always returns null, I cannot find in the Snoop source code how it's done.

See my code:
` Process process = Process.GetProcessesByName("MyWpfApp")[0];`
`HwndSource hwndSource = HwndSource.FromHwnd(process.MainWindowHandle);`
`Window window = hwndSource.RootVisual as Window;`

Thanks

**Comment by batzen** [comment, marked as answer]
Hi,

that's because `HwndSource.FromHwnd` does not work across process boundaries.

To do what Snoop does you have to inject your code into the target process (which requires things like [Snoop.GenericInjector](https://github.com/snoopwpf/snoopwpf/tree/develop/Snoop.GenericInjector) and [Snoop.InjectorLauncher](https://github.com/snoopwpf/snoopwpf/tree/develop/Snoop.InjectorLauncher).
It's complicated and requires quite a lot of code to get it right, and it requires unmanaged C++ code.

**Reply by samberk1** [reply]
Thanks for your answer.
